lib/archethic/beacon_chain/replication_attestation.ex

```diff
@@ -163,49 +159,43 @@ defmodule Archethic.BeaconChain.ReplicationAttestation do
     deserialize_confirmations(rest, nb_replicas_signature, [{position, signature} | acc])
   end
 
+  @doc """
+  Return the index of the node based on the authorized node at this timestamp
+  """
+  @spec get_node_index(Crypto.key(), DateTime.t()) :: non_neg_integer()
```

Comment:
I think this function should be in the P2P module, as it can be useful elsewhere.